repos/system_upgrade/common/actors/inhibitwhenluks/libraries/inhibitwhenluks.py

```diff
@@ -0,0 +1,119 @@
+from leapp import reporting
+from leapp.libraries.common.config.version import get_target_major_version
+from leapp.libraries.stdlib import api
+from leapp.models import CephInfo, DracutModule, LuksDumps, TargetUserSpaceUpgradeTasks, UpgradeInitramfsTasks
+from leapp.reporting import create_report
+
+CLEVIS_RHEL8_DOC_URL = 'https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption_security-hardening#configuring-manual-enrollment-of-volumes-using-tpm2_configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption'  # noqa: E501; pylint: disable=line-too-long
+CLEVIS_RHEL9_DOC_URL = 'https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/security_hardening/configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption_security-hardening#configuring-manual-enrollment-of-volumes-using-tpm2_configuring-automated-unlocking-of-encrypted-volumes-using-policy-based-decryption'  # noqa: E501; pylint: disable=line-too-long
+LUKS2_CONVERT_RHEL8_DOC_URL = 'https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/encrypting-block-devices-using-luks_security-hardening#luks-versions-in-rhel_encrypting-block-devices-using-luks'  # noqa: E501; pylint: disable=line-too-long
+LUKS2_CONVERT_RHEL9_DOC_URL = 'https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/security_hardening/encrypting-block-devices-using-luks_security-hardening#luks-versions-in-rhel_encrypting-block-devices-using-luks'  # noqa: E501; pylint: disable=line-too-long
+FMT_LIST_SEPARATOR = '\n    - '
+
+
+def _at_least_one_tpm_token(luks_dump):
+    return any([token.token_type == "clevis-tpm2" for token in luks_dump.tokens])
+
+
+def check_invalid_luks_devices():
+    target_major_version = get_target_major_version()
+    if target_major_version == '8':
+        clevis_doc_url = CLEVIS_RHEL8_DOC_URL
+        luks2_convert_doc_url = LUKS2_CONVERT_RHEL8_DOC_URL
+    elif target_major_version == '9':
+        clevis_doc_url = CLEVIS_RHEL9_DOC_URL
+        luks2_convert_doc_url = LUKS2_CONVERT_RHEL9_DOC_URL
+    else:
+        create_report([
+            reporting.Title('LUKS encrypted partition detected'),
+            reporting.Summary('Upgrading system with encrypted partitions is not supported'),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Groups([reporting.Groups.BOOT, reporting.Groups.ENCRYPTION]),
+            reporting.Groups([reporting.Groups.INHIBITOR]),
+        ])
+        return
```

Comment:
@danzatt  is there such a difference for 9 -> 10 upgrades? just curious whether the current solution would be broken there or not.  Maybe we discussed that but I do not remember it now, so I am rather asking.